library/alloc/src/collections/btree/map.rs

```diff
@@ -40,30 +40,10 @@ pub(super) const MIN_LEN: usize = node::MIN_LEN_AFTER_SPLIT;
 
 /// An ordered map based on a [B-Tree].
 ///
-/// B-Trees represent a fundamental compromise between cache-efficiency and actually minimizing
-/// the amount of work performed in a search. In theory, a binary search tree (BST) is the optimal
-/// choice for a sorted map, as a perfectly balanced BST performs the theoretical minimum amount of
-/// comparisons necessary to find an element (log<sub>2</sub>n). However, in practice the way this
-/// is done is *very* inefficient for modern computer architectures. In particular, every element
-/// is stored in its own individually heap-allocated node. This means that every single insertion
-/// triggers a heap-allocation, and every single comparison should be a cache-miss. Since these
-/// are both notably expensive things to do in practice, we are forced to, at the very least,
-/// reconsider the BST strategy.
-///
-/// A B-Tree instead makes each node contain B-1 to 2B-1 elements in a contiguous array. By doing
-/// this, we reduce the number of allocations by a factor of B, and improve cache efficiency in
```

Comment:
If you do want to salvage e.g. a brief mention of fewer nodes with more elements being better for reducing heap allocations and cache misses, consider this suggestions from https://github.com/rust-lang/rust/issues/134088#issuecomment-2530463172

> If we do wish to explain the theory behind B-tree performance, we can do it near the end, when we are not crushed for time and space.

Reply:
I've restored the old comments with the suggested fixes from @danielrainer from #134088 in a new background section below everything.